src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyPropertyGenerator.cs

```diff
@@ -0,0 +1,493 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Drawing.Text;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Uno.Extensions;
+using Uno.SourceGeneration;
+using Uno.UI.SourceGenerators.Helpers;
+using Uno.UI.SourceGenerators.XamlGenerator;
+
+namespace Uno.UI.SourceGenerators.DependencyObject
+{
+	public class DependencyPropertyGenerator : SourceGenerator
+	{
+		public override void Execute(SourceGeneratorContext context)
+		{
+			if (PlatformHelper.IsValidPlatform(context))
+			{
+				var visitor = new SerializationMethodsGenerator(context);
+				visitor.Visit(context.Compilation.SourceModule);
```

Comment:
Codacy found an issue: [Refactor this method to add validation of parameter 'context' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5714227)